types/domain_type.go

```diff
@@ -0,0 +1,113 @@
+package types
+
+import (
+	"crypto/sha256"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/pkg/errors"
+)
+
+// NetworkID are intended to separate different SSV networks. A network can have many forks in it.
+type NetworkID [1]byte
+
+var (
+	MainnetNetworkID = NetworkID{0x0}
+	PrimusNetworkID  = NetworkID{0x1}
+	ShifuNetworkID   = NetworkID{0x2}
+	JatoNetworkID    = NetworkID{0x3}
+	JatoV2NetworkID  = NetworkID{0x4}
+)
+
+// DomainType is a unique identifier for signatures, 2 identical pieces of data signed with different domains will result in different sigs
+type DomainType [4]byte
+
+// DomainTypes represent specific forks for specific chains, messages are signed with the domain type making 2 messages from different domains incompatible
+var (
+	GenesisMainnet = DomainType{0x0, 0x0, MainnetNetworkID.Byte(), 0x0}
+	PrimusTestnet  = DomainType{0x0, 0x0, PrimusNetworkID.Byte(), 0x0}
+	ShifuTestnet   = DomainType{0x0, 0x0, ShifuNetworkID.Byte(), 0x0}
+	ShifuV2Testnet = DomainType{0x0, 0x0, ShifuNetworkID.Byte(), 0x1}
+	JatoTestnet    = DomainType{0x0, 0x0, JatoNetworkID.Byte(), 0x1}
+	JatoV2Testnet  = DomainType{0x0, 0x0, JatoV2NetworkID.Byte(), 0x1}
+)
+
+// ForkData is a simple structure holding fork information for a specific chain (and its fork)
+type ForkData struct {
+	// Epoch in which the fork happened
+	Epoch phase0.Epoch
+	// Domain for the new fork
+	Domain DomainType `ssz-size:"4"`
+}
+
+func (n NetworkID) Byte() byte {
+	return n[0]
+}
+
+func (n NetworkID) GetForksData() []*ForkData {
+	switch n {
+	case MainnetNetworkID:
+		return mainnetForks()
+	case PrimusNetworkID:
+		return []*ForkData{{Epoch: 0, Domain: PrimusTestnet}}
+	case JatoNetworkID:
+		return []*ForkData{{Epoch: 0, Domain: JatoTestnet}}
+	case JatoV2NetworkID:
+		return []*ForkData{{Epoch: 0, Domain: JatoV2Testnet}}
+	default:
+		return []*ForkData{}
+	}
+}
+
+func (n NetworkID) DefaultFork() *ForkData {
+	return n.GetForksData()[0]
+}
```

Comment:
In [SIP discussion](https://github.com/bloxapp/SIPs/pull/34#discussion_r1308932572), it's mentioned that the list of forks should be in order. The DefaultFork() method also assumes this. It’s a good idea to make sure this is always the case to avoid any problems later on.

I suggest adding a comment in the `func (n NetworkID) GetForksData() []*ForkData` method to mention that the network fork list should be ordered. It might also be helpful to have a test that checks the forks are listed in order.

also, I'm trying to understand the purpose of the DefaultFork() function. Is it supposed to return the current fork we're on, or is it meant to return the original genesis fork?

Reply:
I'll add it. Thanks :)
Regarding the `DefaultFork` function, initially, it was being used for initialization because the Runner was holding fork info from the network. But this was dropped, and, also, since we want to remove as many functionalities from the spec as possible, I'll remove this function.
Thanks for noticing that.